packages/smooth_app/lib/pages/product/add_basic_details_page.dart

```diff
@@ -125,30 +131,63 @@ class _AddBasicDetailsPageState extends State<AddBasicDetailsPage> {
                     if (!_formKey.currentState!.validate()) {
                       return;
                     }
-                    final bool savedAndRefreshed =
-                        await ProductRefresher().saveAndRefresh(
-                      context: context,
-                      localDatabase: localDatabase,
-                      product: Product(
-                        productName: _productNameController.text,
-                        quantity: _weightController.text,
-                        brands: _brandNameController.text,
-                        barcode: _product.barcode,
+                    final Product inputProduct = Product(
+                      productName: _productNameController.text,
+                      quantity: _weightController.text,
+                      brands: _brandNameController.text,
+                      barcode: _product.barcode,
+                    );
+                    final String uniqueId = const UniqueIdGenerator()
+                        .generateUniqueId(_product.barcode!, 'BASIC_DETAILS');
+                    final BackgroundOtherDetailsInput
+                        backgroundBasicDetailsInput =
+                        BackgroundOtherDetailsInput(
+                      processName: PRODUCT_EDIT_TASK,
+                      uniqueId: uniqueId,
+                      barcode: _product.barcode!,
+                      inputMap: jsonEncode(inputProduct.toJson()),
+                      languageCode: ProductQuery.getLanguage().code,
+                      user: jsonEncode(ProductQuery.getUser().toJson()),
+                      country: ProductQuery.getCountry()!.iso2Code,
+                    );
+                    await TaskManager().addTask(
+                      Task(
+                        data: backgroundBasicDetailsInput.toJson(),
+                        uniqueId: uniqueId,
                       ),
                     );
-                    if (!savedAndRefreshed) {
-                      return;
+
+                    final DaoProduct daoProduct = DaoProduct(localDatabase);
+                    final Product? product = await daoProduct.get(
+                      _product.barcode!,
+                    );
+                    if (product == null) {
+                      daoProduct.put(Product(
```

Comment:
Just for my understand, why do you update `product` in the `else` case and not here?

Reply:
So what I am doing is, I am looking if the product is in the database,
-  if it is not I just create a product and put it in the database so that the user immediately sees the product, 
-  if the product was already in the DB then I just update the changed details about the product and save to DB

Reply:
Could add this in the code?
If I have asked you this question, someone else will also have it ;)